consumer/handler/collection.go

```diff
@@ -0,0 +1,60 @@
+package handler
+
+import (
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// A Collection is set of Handler types, keyed by topic. Only one
+// handler may exists in a Collection per topic, but any number of
+// topics may be contained in the Collection. The handler for a topic
+// can be added via Set, and retrieved via Get. It is not currently
+// possible to remove a topic. Additionally the set of all topics with
+// handlers can be returned via the Topics function. These actions are
+// safe to use in concurrent code.
+type Collection struct {
+	sync.RWMutex
+
+	logger   *logrus.Entry
+	handlers map[string]Handler
+}
+
+// Get returns the Handler associated with a given topic, and a
+// Boolean value indicating if the topic was found at all. It is safe
+// to use Get from concurrent code.
+func (h *Collection) Get(topic string) (handler Handler, ok bool) {
+	h.RLock()
+	handler, ok = h.handlers[topic]
+	h.RUnlock()
+	return
+}
+
+// Topics returns a slice of all the topic names for which the
+// Collection contains a Handler. It is safe to use Topics from
+// concurrent code.
+func (h *Collection) Topics() []string {
+	h.RLock()
+	i := 0
+	count := len(h.handlers)
+	topics := make([]string, count)
+	for t := range h.handlers {
```

Comment:
❓ Why not using the `for i, t := range h.handlers...` syntax?

Reply:
Because handlers is a map, if you use that syntax `i` will be a key and `t` will be its value.

Reply:
Oops, I thought it was a list. Sorry 🙊